bot/code_review_bot/tasks/docupload.py

```diff
@@ -1,23 +1,58 @@
 # -*- coding: utf-8 -*-
+import os
+
 import structlog
 
 from code_review_bot.tasks.base import NoticeTask
 
 logger = structlog.get_logger(__name__)
 
+MAX_LINKS = 20
+
+DOC_LINK = """
+- file [{path}]({doc_url})
+"""
+COMMENT_LINKS_TO_DOC = """
+NOTE: {nb_docs_hint} modified in diff {diff_id}
+
+{pronoun} can be previewed for one week:{doc_urls}
+"""
+
 COMMENT_LINK_TO_DOC = """
-NOTE: Documentation was modified in diff {diff_id}
+NOTE: Several documentation files were modified in diff {diff_id}
 
-It can be previewed [here]({doc_url}) for one week.
+They can be previewed [here]({doc_url}) for one week.
 """
 
 
+def direct_doc_url(path, doc_url, trees):
+    root_doc_url = os.path.dirname(doc_url)
+    filename, _ = os.path.splitext(os.path.basename(path))
+    dir = os.path.dirname(path)
+    for doc_path, dir_path in trees.items():
+        if dir.startswith(dir_path):
+            truncated_dir = dir.replace(dir_path, "", 1)
+            # Forging the link towards the documentation
+            return "/".join(
+                part.strip("/")
+                for part in [root_doc_url, doc_path, truncated_dir, f"{filename}.html"]
+                if part
+            )
```

Comment:
Nit: you could use urljoin maybe or os.path.join to avoid the strip

Reply:
`os.path.join` doesn't work without trimming leading `/`:
```
>>> os.path.join("https://test.com", "folderA", "", "folderB") 
'https://test.com/folderA/folderB'
>>> os.path.join("https://test.com", "/folderA", "", "folderB") 
'/folderA/folderB'
>>> os.path.join("https://test.com", "/folderA", "", "/folderB") 
'/folderB'
```

And `urljoin` doesn't work without ending `/` (and is quite ugly too ahah):
```
>>> urljoin("https://test.com", urljoin("folderA", urljoin("", "folderB"))) 
'https://test.com/folderB'
>>> urljoin(urljoin(urljoin("https://test.com", "folderA"), ""), "folderB") 
'https://test.com/folderB'
```

Do you have any specific idea in mind on how to make this work?

Reply:
You're right, urljoin would still require some changes to the strings (to add "/" when needed), so might as well do what you're doing right now.